website/upgrade-guides/0-15.html.markdown

```diff
@@ -0,0 +1,378 @@
+
+    ```hcl
+    variable "example" {
+      type = list(string)
+    }
+
+    variable "example" {
+      type = map(string)
+    }
+    ```
+
+* In `lifecycle` blocks nested inside `resource` blocks, Terraform previously
+  supported a legacy value `["*"]` for the `ignore_changes` argument, which
+  is removed in Terraform v0.15.
+
+    Instead, use the `all` keyword to indicate that you wish to ignore changes
+    to all of the resource arguments:
+
+    ```hcl
+      lifecycle {
+        ignore_changes = all
+      }
+    ```
+
+* Finally, Terraform v0.11 and earlier required all non-constant expressions
+  to be written using string interpolation syntax, even if the result was
+  not a string. Terraform v0.12 introduced a less confusing syntax where
+  arguments can accept any sort of expression without any special wrapping,
+  and so the interpolation-style syntax has been redundant and deprecated
+  in recent Terraform versions.
+
+  For this particular change we have not made the older syntax invalid, but
+  we do still recommend updating interpolation-only expressions to bare
+  expressions to improve readability:
+
+  ```diff
+  - example = "${var.foo}"
+  + example = var.foo
+  ```
+
+  This only applies to arguments where the value is a single expression without
+  any string concatenation. You must continue to use the `${ ... }` syntax for
+  situations where you are combining string values together into a larger
+  string.
+
+  The `terraform fmt` command can detect and repair simple examples of the
+  legacy interpolation-only syntax, and so we'd recommend running
+  `terraform fmt` on your modules once you've addressed any of the other
+  situations above that could block configuration parsing in order to update
+  your configurations to the typical Terraform language style conventions.
+
+## Alternative Provider Configurations Within Modules
+
+Terraform's provider configuration scheme includes the idea of a "default"
+(unaliased) provider configuration along with zero or more alternative
+(aliased) provider configurations.
+
+The `required_providers` block now has a new field for providers to indicate
+aliased configuration names, replacing the need for an empty "proxy
+configuration block" as a placeholder. In order to declare
+[configuration aliases](/docs/language/modules/develop/providers.html#provider-aliases-within-modules), 
+add the desired names to the `configuration_aliases` argument for the provider
+requirements.
+
+```hcl
+terraform {
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = ">= 2.7.0"
+      configuration_aliases = [ aws.alternate ]
+    }
+  }
+}
+```
+
+Warnings will be emitted now where empty configuration blocks are present but
+no longer required, though they continue to work unchanged in the 0.15 release.
+There are a few cases where existing configurations may return new errors:
+
+* A provider configuration within a module cannot be overridden by supplying a
```

Comment:
The subject of these bullets are inconsistent and I found that made this section a bit hard to follow. I think it would flow better if the first sentence of each bullet was more consistent. I like the last bullet, which is in the form "<thing> is now an error". since it matches your introductory sentence.

How does that sound to you?

Reply:
It looks like @jbardin updated this part, since it was in the commit he'd added. I'm going to run with that updated version for the moment, cause we'll be doing at least a couple more editorial passes of those whole thing before v0.15.0 final. Thanks!